source/apiVolontaria/volunteer/tests/tests_view_CyclesId.py

```diff
@@ -0,0 +1,181 @@
+            reverse(
+                'volunteer:cycles_id',
+                kwargs={'pk': 999},
+            )
+        )
+
+        content = {"detail": "Not found."}
+        self.assertEqual(json.loads(response.content), content)
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_retrieve_cycle(self):
+        """
+        Ensure we can retrieve a cycle.
+        """
+        start_date = self.cycle.start_date.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        end_date = self.cycle.end_date.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+
+        data = {
+            "id": self.cycle.id,
+            "name": self.cycle.name,
+            "start_date": start_date,
+            "end_date": end_date,
+        }
+
+        self.client.force_authenticate(user=self.user)
+
+        response = self.client.get(
+            reverse(
+                'volunteer:cycles_id',
+                kwargs={'pk': self.cycle.id},
+            )
+        )
+
+        self.assertEqual(json.loads(response.content), data)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_update_cycle_with_permission(self):
+        """
+        Ensure we can update a specific cycle.
+        """
+        start_date = self.cycle.start_date.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        end_date = self.cycle.end_date.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+
+        data = {
+            "id": self.cycle.id,
+            "name": "my new_name",
+            "start_date": start_date,
+            "end_date": end_date,
+        }
+
+        data_post = {
+            "name": "my new_name",
+
+        }
+
+        self.client.force_authenticate(user=self.admin)
+
+        response = self.client.patch(
+            reverse(
+                'volunteer:cycles_id',
+                kwargs={'pk': self.cycle.id},
+            ),
+            data_post,
+            format='json',
+        )
+
+        self.assertEqual(json.loads(response.content), data)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_update_cycle_without_permission(self):
+        """
+        Ensure we can't update a specific cycle without permission.
+        """
+        start_date = self.cycle.start_date.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        end_date = self.cycle.end_date.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+
+        data = {
+            "id": self.cycle.id,
+            "name": "my new_name",
+            "start_date": start_date,
+            "end_date": end_date,
+        }
+
+        data_post = {
+            "name": "my new_name",
+
+        }
+
+        self.client.force_authenticate(user=self.user)
+
+        response = self.client.patch(
+            reverse(
+                'volunteer:cycles_id',
+                kwargs={'pk': self.cycle.id},
+            ),
+            data_post,
+            format='json',
+        )
+
+        content = {'detail': "You're not authorized to update a cycle."}
```

Comment:
Same as earlier comment, if we want to keep it formal, no contractions should be used.